kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/InCommitTimestampSuite.scala

```diff
@@ -0,0 +1,150 @@
+import scala.collection.mutable
+import io.delta.kernel.internal.TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED
+import io.delta.kernel.utils.FileStatus;
+
+class InCommitTimestampSuite extends DeltaTableWriteSuiteBase {
+  private def getInCommitTimestamp(engine: Engine, table: Table, version: Long): Optional[Long] = {
+    val logPath = new Path(table.getPath(engine), "_delta_log")
+    val file = FileStatus.of(FileNames.deltaFile(logPath, version), 0, 0)
+    val columnarBatches =
+      engine.getJsonHandler.readJsonFiles(
+        singletonCloseableIterator(file),
+        SingleAction.FULL_SCHEMA,
+        Optional.empty())
+    if (!columnarBatches.hasNext) {
+      return Optional.empty()
+    }
+    val commitInfoVector = columnarBatches.next().getColumnVector(6)
+    for (i <- 0 until commitInfoVector.getSize) {
+      if (!commitInfoVector.isNullAt(i)) {
+        val commitInfo = CommitInfo.fromColumnVector(commitInfoVector, i)
+        if (commitInfo != null && commitInfo.getInCommitTimestamp.isPresent) {
+          return Optional.of(commitInfo.getInCommitTimestamp.get)
+        }
+      }
+    }
+    Optional.empty()
+  }
+
+  test("Enable ICT on commit 0") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val beforeCommitAttemptStartTime = System.currentTimeMillis
+      val table = Table.forPath(engine, tablePath)
+
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = true, IN_COMMIT_TIMESTAMPS_ENABLED, "true", true)
+
+      val ver0Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assert(getInCommitTimestamp(engine, table, 0).get >= beforeCommitAttemptStartTime)
+      assert(ver0Snapshot.getProtocol.getWriterFeatures.contains("inCommitTimestamp-preview"))
+    }
+  }
+
+  test("Create a non-inCommitTimestamp table and then enable timestamp") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
+
+      val txn1 = txnBuilder
+        .withSchema(engine, testSchema)
+        .build(engine)
+
+      txn1.commit(engine, emptyIterable())
+
+      val ver0Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver0Snapshot, TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED, false)
+      assert(!ver0Snapshot.getProtocol.getWriterFeatures.contains("inCommitTimestamp-preview"))
+      assert(!getInCommitTimestamp(engine, table, 0).isPresent)
+
+      val beforeCommitAttemptStartTime = System.currentTimeMillis
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = false, IN_COMMIT_TIMESTAMPS_ENABLED, "true", true)
+
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assert(ver1Snapshot.getProtocol.getWriterFeatures.contains("inCommitTimestamp-preview"))
+      assert(
+        getInCommitTimestamp(engine, table, 1).get >= beforeCommitAttemptStartTime)
```

Comment:
It would be good to eventually have a version of transaction builder that takes in a Clock so that this can be tested more thoroughly.

Reply:
Yes. I will add it in the next PR. Or do you think I should add the Clock in this PR?

Reply:
I just saw that you added it in this PR: https://github.com/delta-io/delta/pull/3282. No need to shift it to this one.